docs/xchain.js

```diff
@@ -388,60 +605,333 @@ function validatePact() {
+      document.getElementById('gas-price'),
+    ].forEach(e => {
+      e.addEventListener('blur', e => {
+        fillSigData();
+      });
+    });
+
+    document.getElementById('click-to-copy').addEventListener('click', e => {
+      e.preventDefault();
+      const clickToCopyElement = e.target;
+      const textArea = document.getElementById('sig-data');
+
+      // Select the text field
+      textArea.select();
+      textArea.setSelectionRange(0, 99999); // For mobile devices
+
+      // Copy the text inside the text field
+      navigator.clipboard.writeText(textArea.value);
+      clickToCopyElement.classList.remove('td-underline');
+      clickToCopyElement.innerText = 'Copied!';
+      setTimeout(() => {
+        clickToCopyElement.innerText = 'Click To Copy';
+        clickToCopyElement.classList.add('td-underline');
+      }, 1000);
+    });
+
+    document
+      .getElementById('gas-price')
+      .addEventListener('input', onInputGasPrice);
+
+    document
+      .getElementById('gas-limit')
+      .addEventListener('input', onInputGasLimit);
   },
-  false,
+
+  false
 );
+
+async function validateGasPayer() {
+  await getCoinDetailsOfGasPayer();
+  clearError();
+  if (isAccountEligibleForGasPayment()) {
+    const sigDataMessage = document.getElementById('sig-data-message');
+    const signedTransactionWrapper = document.getElementById(
+      'signed-transaction-wrapper'
+    );
+    const signedTransaction = document.getElementById('signed-transaction');
+    if (window.isGasStation(State.gasPayerAccountDetails)) {
+      sigDataMessage.classList.add('hidden');
+      signedTransactionWrapper.classList.add('hidden');
+      signedTransaction.value = '';
+      clearError();
+    } else {
+      sigDataMessage.classList.remove('hidden');
+      signedTransactionWrapper.classList.remove('hidden');
+    }
+  }
+}
+
+async function fillSigData() {
+  enableSubmit();
+  if (!window.proof) {
+    window.proof = await getProof();
+  }
+  const pactId = State.pactId;
+  const networkId = State.networkId;
+
+  const m = Pact.lang.mkMeta(
+    State.gasPayer,
+    State.targetChainId,
+    State.gasPrice,
+    State.gasLimit,
+    createTime(),
+    28800
+  );
+
+  const keys = new Keys().add(State.senderPublicKey, 'coin.GAS'),
+    c = Pact.simple.cont.createCommand(
+      keys,
+      `transfer.chainweb ${State.requestKey} ${new Date().toLocaleString()}`,
+      1,
+      pactId,
+      false,
+      undefined,
+      m,
+      proof,
+      networkId
+    ).cmds[0];
+
+  c.sigs = keys.getSigsObject();
+
+  setTimeout(
+    () =>
+      (document.getElementById('sig-data').value = JSON.stringify(c, null, 2))
+  );
+}
+
+function isGasStation(res) {
+  return JSON.stringify(res.result.data?.guard).includes('gas-only');
+}
+
+function isBalanceSufficient(gasPrice, gasLimit, res) {
+  try {
+    return gasPrice * gasLimit <= res.result.data.balance;
+  } catch (error) {
+    return false;
+  }
+}
+
+function isSingleSig(res) {
```

Comment:
This doesn't work for multisigs that only require 1 sig. But maybe that's ok. Perhaps add a comment